{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import csv\n",
    "import pandas as pd\n",
    "import glob\n",
    "import math\n",
    "from pathlib import Path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "hpc_codes_dir = \"/Volumes/Huitian/Projects/T_Cell_ChIP/codes_hpc\"\n",
    "\n",
    "#--- Get list of sra files, retrive Run IDs\n",
    "sra_dir = \"/Volumes/Huitian/Projects/T_Cell_ChIP/202012_ChIP/1_SRA_Run_Table_simplified\"\n",
    "sra_files = glob.glob(\"%s/*.csv\"%sra_dir)\n",
    "\n",
    "srr_list = []\n",
    "srr_layout = []\n",
    "for file in sra_files:\n",
    "    srr_list += pd.read_csv(file)['Run'].tolist()\n",
    "    srr_layout += pd.read_csv(file)['LibraryLayout'].tolist()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 0_0_fastq-dump"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "#--- Write Hpc script\n",
    "out_file = \"0_0_fastq-dump\"\n",
    "out_dir = hpc_codes_dir + \"/\" + out_file\n",
    "Path(out_dir).mkdir(parents=True, exist_ok=True)\n",
    "\n",
    "hpc_wkdir = \"/gpfs/group/pipkin/hdiao/T_Cell_ChIP/0_fastq\"\n",
    "# Write one script for every 16 files\n",
    "for i in range(0, math.ceil(len(srr_list)/16)):\n",
    "    i_outname = \"%s/%s-%s.sh\"%(out_dir, out_file,i)\n",
    "    with open(i_outname, \"w\") as fout:\n",
    "        wfout = csv.writer(fout, delimiter=\"\\t\",lineterminator='\\n')\n",
    "        wfout.writerow([\"#!/bin/bash\"])\n",
    "        wfout.writerow([\"#SBATCH --nodes=1\"])\n",
    "        wfout.writerow([\"#SBATCH --ntasks=8\"])\n",
    "        wfout.writerow([\"#SBATCH --mem=16gb\"])\n",
    "        wfout.writerow([])\n",
    "        wfout.writerow([\"module load sra-tools\"])\n",
    "        wfout.writerow([\"cd %s\"%hpc_wkdir])\n",
    "        wfout.writerow([])\n",
    "        \n",
    "        for j in range(i*16,min((i+1) * 16, len(srr_list))):\n",
    "            j_srr = srr_list[j]\n",
    "            if ((j+1) % 8 != 0) and (j < min((i+1) * 16, len(srr_list))-1):\n",
    "                wfout.writerow([\"fastq-dump -I --split-files %s &\"%j_srr])\n",
    "            elif j < min((i+1) * 16, len(srr_list))-1:\n",
    "                wfout.writerow([\"fastq-dump -I --split-files %s \"%j_srr])\n",
    "                wfout.writerow([\"wait\"])\n",
    "            else:\n",
    "                wfout.writerow([\"fastq-dump -I --split-files %s \"%j_srr])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1_0_trim_alignment_convert_filterBlacklist"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "#--- Write Hpc script\n",
    "out_file = \"1_0_trim_alignment_flb\"\n",
    "out_dir = hpc_codes_dir + \"/\" + out_file\n",
    "Path(out_dir).mkdir(parents=True, exist_ok=True)\n",
    "\n",
    "#--- HPC setup\n",
    "hpc_wkdir = \"/gpfs/group/pipkin/hdiao/T_Cell_ChIP/1_bowtie2\"\n",
    "hpc_inputdir = \"/gpfs/group/pipkin/hdiao/T_Cell_ChIP/0_fastq\"\n",
    "bowtie2_index = \"/gpfs/group/pipkin/hdiao/ref_resources/mm/release102/GRCm38\"\n",
    "blacklisted_bed = \"/gpfs/group/pipkin/hdiao/ref_resources/mm/mm10_blacklisted_2016_nochr.bed\"\n",
    "\n",
    "# Write one script for every 4 files\n",
    "for i in range(0, len(srr_list)):\n",
    "    i_outname = \"%s/%s-%s.sh\"%(out_dir, out_file,i)\n",
    "    srr_i = srr_list[i]\n",
    "    srr_layout_i = srr_layout[i]\n",
    "    \n",
    "    with open(i_outname, \"w\") as fout:\n",
    "        wfout = csv.writer(fout, delimiter=\"\\t\",lineterminator='\\n')\n",
    "        wfout.writerow([\"#!/bin/bash\"])\n",
    "        wfout.writerow([\"#SBATCH --nodes=1\"])\n",
    "        wfout.writerow([\"#SBATCH --ntasks=16\"])\n",
    "        wfout.writerow([\"#SBATCH --mem=8gb\"])\n",
    "        wfout.writerow([])\n",
    "        wfout.writerow([\"module load fastqc\"])\n",
    "        wfout.writerow([\"module load trimgalore\"])\n",
    "        wfout.writerow([\"module load samtools\"])\n",
    "        wfout.writerow([\"module load bowtie2\"])\n",
    "        wfout.writerow([\"module load bedtools\"])\n",
    "        \n",
    "        ### Fastqc untrimmed\n",
    "        wfout.writerow([])\n",
    "        wfout.writerow([\"### Fastqc for untrimmed files\"])\n",
    "        wfout.writerow([\"cd %s\"%hpc_inputdir])\n",
    "        if srr_layout_i == \"PAIRED\":\n",
    "            wfout.writerow([\"fastq_untrimmed_1=%s_1.fastq\"%srr_i])\n",
    "            wfout.writerow([\"fastq_untrimmed_2=%s_2.fastq\"%srr_i])\n",
    "            wfout.writerow([\"fastqc $fastq_untrimmed_1\"])\n",
    "            wfout.writerow([\"fastqc $fastq_untrimmed_2\"])\n",
    "        else:\n",
    "            wfout.writerow([\"fastq_untrimmed_1=%s_1.fastq\"%srr_i])\n",
    "            wfout.writerow([\"fastqc $fastq_untrimmed_1\"])\n",
    "        \n",
    "        ### Trim galore\n",
    "        wfout.writerow([])\n",
    "        wfout.writerow([\"### Trim Galore\"])\n",
    "        if srr_layout_i == \"PAIRED\":\n",
    "            wfout.writerow([\"trim_galore --paired --length 24 --stringency 3 $fastq_untrimmed_1 $fastq_untrimmed_2\"])\n",
    "            wfout.writerow([\"trim_fastq_end1=%s/%s_1_val_1.fq\"%(hpc_inputdir, srr_i)])\n",
    "            wfout.writerow([\"trim_fastq_end2=%s/%s_2_val_2.fq\"%(hpc_inputdir, srr_i)])\n",
    "        else:        \n",
    "            wfout.writerow([\"trim_galore --length 24 --stringency 3 $fastq_untrimmed_1\"])  \n",
    "            wfout.writerow([\"trim_fastq_end1=%s/%s_1_trimmed.fq\"%(hpc_inputdir, srr_i)])\n",
    "            \n",
    "        ### Trimmed file fastqc\n",
    "        wfout.writerow([])\n",
    "        wfout.writerow([\"### Fastqc for trimmed files\"])\n",
    "        if srr_layout_i == \"PAIRED\":\n",
    "            wfout.writerow([\"fastqc $trim_fastq_end1\"])\n",
    "            wfout.writerow([\"fastqc $trim_fastq_end2\"])\n",
    "        else:\n",
    "            wfout.writerow([\"fastqc $trim_fastq_end1\"])\n",
    "        \n",
    "        ### Bowtie2 alignment\n",
    "        wfout.writerow([])\n",
    "        wfout.writerow([\"### Bowtie2 alignment\"])\n",
    "        wfout.writerow([\"cd %s\"%hpc_wkdir])\n",
    "        wfout.writerow([\"bowtie2_index=%s\"%bowtie2_index])\n",
    "        wfout.writerow([\"sam_name=%s.sam\"%srr_i])\n",
    "        if srr_layout_i == \"PAIRED\":\n",
    "            wfout.writerow([\"bowtie2 -p 16 -x $bowtie2_index -X 1000 --fr -1 $trim_fastq_end1 -2 $trim_fastq_end2 -S $sam_name\"])\n",
    "        else:\n",
    "            wfout.writerow([\"bowtie2 -p 16 -x $bowtie2_index -U $trim_fastq_end1 -S $sam_name\"])\n",
    "        \n",
    "        ### Convert & Sort & Filter\n",
    "        wfout.writerow([])\n",
    "        wfout.writerow([\"### Convert/sort/filter\"])\n",
    "        wfout.writerow([\"bam_name=%s.bam\"%srr_i])\n",
    "        wfout.writerow([\"bam_name_srt=%s_srt.sam\"%srr_i])\n",
    "        wfout.writerow([\"sam_name_srt_dupr=%s_srt_dupr.sam\"%srr_i])\n",
    "        wfout.writerow([\"bam_name_srt_dupr=%s_srt_dupr.bam\"%srr_i])\n",
    "        wfout.writerow([\"flb_bam_name=%s_srt_dupr_flb.bam\"%srr_i])\n",
    "        wfout.writerow([\"blacklist_bed=%s\"%blacklisted_bed])\n",
    "        wfout.writerow([])\n",
    "        wfout.writerow([\"samtools view -bS $sam_name > $bam_name\"])\n",
    "        wfout.writerow([\"samtools sort $bam_name -o $bam_name_srt\"])\n",
    "        wfout.writerow([\"samtools rmdup -S $bam_name_srt $sam_name_srt_dupr\"])\n",
    "        wfout.writerow([\"samtools view -bS $sam_name_srt_dupr > $bam_name_srt_dupr\"])\n",
    "        wfout.writerow([\"bedtools intersect -abam $bam_name_srt_dupr -b $blacklist_bed -v > $flb_bam_name\"])\n",
    "        \n",
    "        \n",
    "        ### Delete intermediate files\n",
    "        wfout.writerow([])\n",
    "        wfout.writerow([\"### Remove intermediate files\"])\n",
    "        wfout.writerow([\"filesize=$(stat -c%s $flb_bam_name)\"])\n",
    "        wfout.writerow([\"if (( filesize > 10000 )) \"])\n",
    "        wfout.writerow([\"then\"])\n",
    "        wfout.writerow([\"    rm $sam_name\"])\n",
    "        wfout.writerow([\"    rm $bam_name\"])\n",
    "        wfout.writerow([\"    rm $bam_name_srt\"])\n",
    "        wfout.writerow([\"    rm $sam_name_srt_dupr\"])\n",
    "        wfout.writerow([\"    rm $bam_name_srt_dupr\"])\n",
    "        wfout.writerow([\"    rm $trim_fastq_end1\"])\n",
    "        if srr_layout_i == \"PAIRED\":\n",
    "            wfout.writerow([\"    rm $trim_fastq_end2\"])\n",
    "        wfout.writerow([\"fi\"])\n",
    "                \n",
    "        "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1_0_trim_alignment_convert_filterBlacklist\n",
    "**Redo for failed, try without trimming**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "failed_alignment = pd.read_csv(\"/Volumes/Huitian/Projects/\\\n",
    "T_Cell_ChIP/codes_hpc/1_0_trim_alignment_flb_check/failed_alignments.csv\")\n",
    "failed_alignment_run = failed_alignment['Run'].tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "#--- Write Hpc script\n",
    "out_file = \"1_0_trim_alignment_flb_redo\"\n",
    "out_dir = hpc_codes_dir + \"/\" + out_file\n",
    "Path(out_dir).mkdir(parents=True, exist_ok=True)\n",
    "\n",
    "#--- HPC setup\n",
    "hpc_wkdir = \"/gpfs/group/pipkin/hdiao/T_Cell_ChIP/1_bowtie2\"\n",
    "hpc_inputdir = \"/gpfs/group/pipkin/hdiao/T_Cell_ChIP/0_fastq\"\n",
    "bowtie2_index = \"/gpfs/group/pipkin/hdiao/ref_resources/mm/release102/GRCm38\"\n",
    "blacklisted_bed = \"/gpfs/group/pipkin/hdiao/ref_resources/mm/mm10_blacklisted_2016_nochr.bed\"\n",
    "srr_list = list(set(srr_list) & set(failed_alignment_run))\n",
    "\n",
    "# Write one script for every 4 files\n",
    "for i in range(0, len(srr_list)):\n",
    "    i_outname = \"%s/%s-%s.sh\"%(out_dir, out_file,i)\n",
    "    srr_i = srr_list[i]\n",
    "    srr_layout_i = srr_layout[i]\n",
    "    \n",
    "    with open(i_outname, \"w\") as fout:\n",
    "        wfout = csv.writer(fout, delimiter=\"\\t\",lineterminator='\\n')\n",
    "        wfout.writerow([\"#!/bin/bash\"])\n",
    "        wfout.writerow([\"#SBATCH --nodes=1\"])\n",
    "        wfout.writerow([\"#SBATCH --ntasks=16\"])\n",
    "        wfout.writerow([\"#SBATCH --mem=8gb\"])\n",
    "        wfout.writerow([])\n",
    "        wfout.writerow([\"module load fastqc\"])\n",
    "        wfout.writerow([\"module load trimgalore\"])\n",
    "        wfout.writerow([\"module load samtools\"])\n",
    "        wfout.writerow([\"module load bowtie2\"])\n",
    "        wfout.writerow([\"module load bedtools\"])\n",
    "        \n",
    "        ### Fastqc untrimmed\n",
    "        wfout.writerow([])\n",
    "        wfout.writerow([\"### Fastqc for untrimmed files\"])\n",
    "        if srr_layout_i == \"PAIRED\":\n",
    "            wfout.writerow([\"fastq_untrimmed_1=%s/%s_1.fastq\"%(hpc_inputdir,srr_i)])\n",
    "            wfout.writerow([\"fastq_untrimmed_2=%s/%s_2.fastq\"%(hpc_inputdir,srr_i)])\n",
    "            wfout.writerow([\"#fastqc $fastq_untrimmed_1\"])\n",
    "            wfout.writerow([\"#fastqc $fastq_untrimmed_2\"])\n",
    "        else:\n",
    "            wfout.writerow([\"fastq_untrimmed_1=%s/%s_1.fastq\"%(hpc_inputdir,srr_i)])\n",
    "            wfout.writerow([\"#fastqc $fastq_untrimmed_1\"])\n",
    "        \n",
    "        \n",
    "        ### Bowtie2 alignment\n",
    "        wfout.writerow([])\n",
    "        wfout.writerow([\"### Bowtie2 alignment\"])\n",
    "        wfout.writerow([\"cd %s\"%hpc_wkdir])\n",
    "        wfout.writerow([\"bowtie2_index=%s\"%bowtie2_index])\n",
    "        wfout.writerow([\"sam_name=%s.sam\"%srr_i])\n",
    "        if srr_layout_i == \"PAIRED\":\n",
    "            wfout.writerow([\"bowtie2 -p 16 -x $bowtie2_index -X 1000 --fr -1 $fastq_untrimmed_1 -2 $fastq_untrimmed_2 -S $sam_name\"])\n",
    "        else:\n",
    "            wfout.writerow([\"bowtie2 -p 16 -x $bowtie2_index -U $fastq_untrimmed_1 -S $sam_name\"])\n",
    "        \n",
    "        ### Convert & Sort & Filter\n",
    "        wfout.writerow([])\n",
    "        wfout.writerow([\"### Convert/sort/filter\"])\n",
    "        wfout.writerow([\"bam_name=%s.bam\"%srr_i])\n",
    "        wfout.writerow([\"bam_name_srt=%s_srt.sam\"%srr_i])\n",
    "        wfout.writerow([\"sam_name_srt_dupr=%s_srt_dupr.sam\"%srr_i])\n",
    "        wfout.writerow([\"bam_name_srt_dupr=%s_srt_dupr.bam\"%srr_i])\n",
    "        wfout.writerow([\"flb_bam_name=%s_srt_dupr_flb.bam\"%srr_i])\n",
    "        wfout.writerow([\"blacklist_bed=%s\"%blacklisted_bed])\n",
    "        wfout.writerow([])\n",
    "        wfout.writerow([\"samtools view -bS $sam_name > $bam_name\"])\n",
    "        wfout.writerow([\"samtools sort $bam_name -o $bam_name_srt\"])\n",
    "        wfout.writerow([\"samtools rmdup -S $bam_name_srt $sam_name_srt_dupr\"])\n",
    "        wfout.writerow([\"samtools view -bS $sam_name_srt_dupr > $bam_name_srt_dupr\"])\n",
    "        wfout.writerow([\"bedtools intersect -abam $bam_name_srt_dupr -b $blacklist_bed -v > $flb_bam_name\"])\n",
    "        \n",
    "        \n",
    "        ### Delete intermediate files\n",
    "        wfout.writerow([])\n",
    "        wfout.writerow([\"### Remove intermediate files\"])\n",
    "        wfout.writerow([\"filesize_preFlb=$(stat -c%s $bam_name_srt_dupr)\"])\n",
    "        wfout.writerow([\"filesize=$(stat -c%s $flb_bam_name)\"])\n",
    "        wfout.writerow([\"echo $filesize_preFlb $filesize >> %s_bamSizes_pre_post_flb.txt\"%srr_i])\n",
    "        wfout.writerow([\"if (( filesize > 10000 )) \"])\n",
    "        wfout.writerow([\"then\"])\n",
    "        wfout.writerow([\"    rm $sam_name\"])\n",
    "        wfout.writerow([\"    rm $bam_name\"])\n",
    "        wfout.writerow([\"    rm $bam_name_srt\"])\n",
    "        wfout.writerow([\"    rm $sam_name_srt_dupr\"])\n",
    "        wfout.writerow([\"    rm $bam_name_srt_dupr\"])\n",
    "        wfout.writerow([\"fi\"])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1_0_trim_alignment_convert_filterBlacklist\n",
    "**Redo for solid sequences, try with / without trimming**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "#--- Get list of sra files, retrive Run IDs\n",
    "sra_dir = \"/Volumes/Huitian/Projects/T_Cell_ChIP/202012_ChIP/1_SRA_Run_Table_simplified\"\n",
    "solid_file = sra_dir + \"/2015_IMMUNITY_Martinez_simplified.csv\"\n",
    "srr_list = pd.read_csv(solid_file)['Run'].tolist()\n",
    "srr_layout = pd.read_csv(solid_file)['LibraryLayout'].tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "#--- Write Hpc script\n",
    "out_file = \"1_0_trim_alignment_flb_solid\"\n",
    "out_dir = hpc_codes_dir + \"/\" + out_file\n",
    "Path(out_dir).mkdir(parents=True, exist_ok=True)\n",
    "\n",
    "#--- HPC setup\n",
    "hpc_wkdir = \"/gpfs/group/pipkin/hdiao/T_Cell_ChIP/1_bowtie2\"\n",
    "hpc_inputdir = \"/gpfs/group/pipkin/hdiao/T_Cell_ChIP/0_fastq\"\n",
    "bowtie2_index = \"/gpfs/group/pipkin/hdiao/ref_resources/mm/release102/GRCm38\"\n",
    "blacklisted_bed = \"/gpfs/group/pipkin/hdiao/ref_resources/mm/mm10_blacklisted_2016_nochr.bed\"\n",
    "\n",
    "# Write one script for every 4 files\n",
    "for i in range(0, len(srr_list)):\n",
    "    i_outname = \"%s/%s-%s.sh\"%(out_dir, out_file,i)\n",
    "    srr_i = srr_list[i]\n",
    "    srr_layout_i = srr_layout[i]\n",
    "    \n",
    "    with open(i_outname, \"w\") as fout:\n",
    "        wfout = csv.writer(fout, delimiter=\"\\t\",lineterminator='\\n')\n",
    "        wfout.writerow([\"#!/bin/bash\"])\n",
    "        wfout.writerow([\"#SBATCH --nodes=1\"])\n",
    "        wfout.writerow([\"#SBATCH --ntasks=16\"])\n",
    "        wfout.writerow([\"#SBATCH --mem=8gb\"])\n",
    "        wfout.writerow([])\n",
    "        wfout.writerow([\"module load fastqc\"])\n",
    "        wfout.writerow([\"module load trimgalore\"])\n",
    "        wfout.writerow([\"module load samtools\"])\n",
    "        wfout.writerow([\"module load bowtie2\"])\n",
    "        wfout.writerow([\"module load bedtools\"])\n",
    "        \n",
    "        ### Fastqc untrimmed\n",
    "        wfout.writerow([])\n",
    "        wfout.writerow([\"### Fastqc for untrimmed files\"])\n",
    "        wfout.writerow([\"cd %s\"%hpc_inputdir])\n",
    "        if srr_layout_i == \"PAIRED\":\n",
    "            wfout.writerow([\"fastq_untrimmed_1=%s/%s_1_solidCvt.fastq\"%(hpc_inputdir, srr_i)])\n",
    "            wfout.writerow([\"fastq_untrimmed_2=%s/%s_2_solidCvt.fastq\"%(hpc_inputdir, srr_i)])\n",
    "            wfout.writerow([\"fastqc $fastq_untrimmed_1\"])\n",
    "            wfout.writerow([\"fastqc $fastq_untrimmed_2\"])\n",
    "        else:\n",
    "            wfout.writerow([\"fastq_untrimmed_1=%s/%s_1_solidCvt.fastq\"%(hpc_inputdir, srr_i)])\n",
    "            wfout.writerow([\"fastqc $fastq_untrimmed_1\"])\n",
    "        \n",
    "        ### Trim galore\n",
    "        wfout.writerow([])\n",
    "        wfout.writerow([\"### Trim Galore\"])\n",
    "        if srr_layout_i == \"PAIRED\":\n",
    "            wfout.writerow([\"trim_galore --paired --length 24 --stringency 3 $fastq_untrimmed_1 $fastq_untrimmed_2\"])\n",
    "            wfout.writerow([\"trim_fastq_end1=%s/%s_1_solidCvt_val_1.fq\"%(hpc_inputdir, srr_i)])\n",
    "            wfout.writerow([\"trim_fastq_end2=%s/%s_2_solidCvt_val_2.fq\"%(hpc_inputdir, srr_i)])\n",
    "        else:        \n",
    "            wfout.writerow([\"trim_galore --length 24 --stringency 3 $fastq_untrimmed_1\"])  \n",
    "            wfout.writerow([\"trim_fastq_end1=%s/%s_1_solidCvt_trimmed.fq\"%(hpc_inputdir, srr_i)])\n",
    "            \n",
    "        ### Trimmed file fastqc\n",
    "        wfout.writerow([])\n",
    "        wfout.writerow([\"### Fastqc for trimmed files\"])\n",
    "        if srr_layout_i == \"PAIRED\":\n",
    "            wfout.writerow([\"fastqc $trim_fastq_end1\"])\n",
    "            wfout.writerow([\"fastqc $trim_fastq_end2\"])\n",
    "        else:\n",
    "            wfout.writerow([\"fastqc $trim_fastq_end1\"])\n",
    "        \n",
    "        ### Test if trimming is successful\n",
    "        wfout.writerow([])\n",
    "        wfout.writerow([\"### Test if trimming is successful\"])\n",
    "        if srr_layout_i == \"PAIRED\":\n",
    "            wfout.writerow([\"filesize=$(stat -c%s $trim_fastq_end2)\"])\n",
    "        else:\n",
    "            wfout.writerow([\"filesize=$(stat -c%s $trim_fastq_end1)\"])\n",
    "        # If trimming did not succeed, use the original files\n",
    "        wfout.writerow([\"if (( filesize < 10000 )) \"])\n",
    "        wfout.writerow([\"then\"])\n",
    "        wfout.writerow([\"    trim_fastq_end1=$fastq_untrimmed_1\"])\n",
    "        if srr_layout_i == \"PAIRED\":\n",
    "            wfout.writerow([\"    trim_fastq_end2=$fastq_untrimmed_2\"])\n",
    "        wfout.writerow([\"fi\"])        \n",
    "        \n",
    "        ### Bowtie2 alignment\n",
    "        wfout.writerow([])\n",
    "        wfout.writerow([\"### Bowtie2 alignment\"])\n",
    "        wfout.writerow([\"cd %s\"%hpc_wkdir])\n",
    "        wfout.writerow([\"bowtie2_index=%s\"%bowtie2_index])\n",
    "        wfout.writerow([\"sam_name=%s.sam\"%srr_i])\n",
    "        if srr_layout_i == \"PAIRED\":\n",
    "            wfout.writerow([\"bowtie2 -p 16 -x $bowtie2_index -X 1000 --fr -1 $trim_fastq_end1 -2 $trim_fastq_end2 -S $sam_name\"])\n",
    "        else:\n",
    "            wfout.writerow([\"bowtie2 -p 16 -x $bowtie2_index -U $trim_fastq_end1 -S $sam_name\"])\n",
    "        \n",
    "        ### Convert & Sort & Filter\n",
    "        wfout.writerow([])\n",
    "        wfout.writerow([\"### Convert/sort/filter\"])\n",
    "        wfout.writerow([\"bam_name=%s.bam\"%srr_i])\n",
    "        wfout.writerow([\"bam_name_srt=%s_srt.sam\"%srr_i])\n",
    "        wfout.writerow([\"sam_name_srt_dupr=%s_srt_dupr.sam\"%srr_i])\n",
    "        wfout.writerow([\"bam_name_srt_dupr=%s_srt_dupr.bam\"%srr_i])\n",
    "        wfout.writerow([\"flb_bam_name=%s_srt_dupr_flb.bam\"%srr_i])\n",
    "        wfout.writerow([\"blacklist_bed=%s\"%blacklisted_bed])\n",
    "        wfout.writerow([])\n",
    "        wfout.writerow([\"samtools view -bS $sam_name > $bam_name\"])\n",
    "        wfout.writerow([\"samtools sort $bam_name -o $bam_name_srt\"])\n",
    "        wfout.writerow([\"samtools rmdup -S $bam_name_srt $sam_name_srt_dupr\"])\n",
    "        wfout.writerow([\"samtools view -bS $sam_name_srt_dupr > $bam_name_srt_dupr\"])\n",
    "        wfout.writerow([\"bedtools intersect -abam $bam_name_srt_dupr -b $blacklist_bed -v > $flb_bam_name\"])\n",
    "        \n",
    "        \n",
    "        ### Delete intermediate files\n",
    "        wfout.writerow([])\n",
    "        wfout.writerow([\"### Remove intermediate files\"])\n",
    "        wfout.writerow([\"filesize_preFlb=$(stat -c%s $bam_name_srt_dupr)\"])\n",
    "        wfout.writerow([\"filesize=$(stat -c%s $flb_bam_name)\"])\n",
    "        wfout.writerow([\"echo $filesize_preFlb $filesize >> %s_bamSizes_pre_post_flb.txt\"%srr_i])\n",
    "        wfout.writerow([\"if (( filesize > 10000 )) \"])\n",
    "        wfout.writerow([\"then\"])\n",
    "        wfout.writerow([\"    rm $sam_name\"])\n",
    "        wfout.writerow([\"    rm $bam_name\"])\n",
    "        wfout.writerow([\"    rm $bam_name_srt\"])\n",
    "        wfout.writerow([\"    rm $sam_name_srt_dupr\"])\n",
    "        wfout.writerow([\"    rm $bam_name_srt_dupr\"])\n",
    "        wfout.writerow([\"    rm $trim_fastq_end1\"])\n",
    "        if srr_layout_i == \"PAIRED\":\n",
    "            wfout.writerow([\"    rm $trim_fastq_end2\"])\n",
    "        wfout.writerow([\"fi\"])\n",
    "                \n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
